core/src/main/scala/com/criteo/cuttle/Authentication.scala

```diff
@@ -0,0 +1,89 @@
+package com.criteo.cuttle
+
+import java.util.Base64
+
+import lol.http._
+
+import scala.util.Try
+
+/**
+  * Trait describing authentication methods
+  */
+trait Authenticator {
+  /**
+    *
+    * @param wrappedAuthedService
+    * @return
+    */
+  def apply(wrappedAuthedService: (User) => Service): Service = (request: Request) => {
+    authenticate(request)
+      .fold(user => wrappedAuthedService(user)(request), identity)
+  }
+
+  /**
+    * Performs authentication on request.
+    * @param r the request to be authenticated
+    * @return either an authenticated user or a response.
+    */
+  def authenticate(r: Request): Either[User, Response]
+}
+
+case class User(userName: String)
+
+case object NoAuth extends Authenticator {
+  /**
+    * Authenticated any user as Guest.
+    * @param r request to be authenticated
+    * @return Authenticated user
+    */
+  override def authenticate(r: Request): Either[User, Response] = Left(User("Guest"))
+}
+
+/**
+  * Implementation of the HTTP Basic auth.
+  * @param credentialsValidator method to validate credentials.
+  */
+case class BasicAuth(credentialsValidator : ((String,String)) => Boolean) extends Authenticator {
+
+  val scheme = "Basic"
+  val unauthorizedResponse = Response(401).addHeaders(h"WWW-Authenticate" -> h"""Basic realm="User Visible Realm"""")
```

Comment:
Please change the realm for something more relevant (or make it configurable with a default value in the BasicAuth constructor).